### PR TITLE
test: Speed up auth tests

### PR DESCRIPTION
**Pull request by jkosh44**
The auth tests sometimes flake due to an expired authentication.

The auth tests start a fake materialize server that responds to Frontegg requests with generated claims. The tests use a constant, EXPIRES_IN_SECS, that is passed into the fake test server, so that all claims are returned with an exp field set to the current time minus EXPIRES_IN_SECS. The exp field is The time at which the claims expire, represented in seconds since the Unix epoch. Essentially EXPIRES_IN_SECS represents how many seconds the claims are valid as of when they're created.

The tests also use a constant, REFRESH_BEFORE_SECS, that determines how many seconds before the claims expire to start refreshing the claims. Essentially, we attempt to refresh a token REFRESH_BEFORE_SECS seconds before it expires.

The test flake happens when the refresh takes long enough that by the time the refresh is complete the token has already expired. As a result, as long as REFRESH_BEFORE_SECS <= EXPIRES_IN_SECS, the flakiness is completely determined by REFRESH_BEFORE_SECS, since that is the amount of time we have to complete the refresh.

This commit lowers EXPIRES_IN_SECS from 40 to 20 (REFRESH_BEFORE_SECS is set to 20) to decrease the time we wait for a token refresh without affecting test flakiness. A better more robust fix would use deterministic clocks to avoid all flakes.

Works towards resolving #22145

### Motivation
This PR fixes a recognized bug.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

**Comment by jkosh44** [comment]
This should ideally lower the time that `wait_for_refresh` takes to ~0 seconds (really however long a refresh takes) since we are constantly refreshing a token immediately after getting it. Additionally it shouldn't reduce test flakiness since we still have 20 seconds from the time a refresh starts to the time a token expires.

Locally I see the following speedups:
```
test_auth_expiry:
  main: 1 min 50 sec
  PR: 30 sec
test_auth_admin_superuser_revoked (formally test_auth_admin_show_is_superuser):
  main: 49 sec
  PR: 9 sec
```

File: src/environmentd/tests/auth.rs [hunk 1]
```diff
@@ -810,8 +810,10 @@ fn test_auth_expiry() {
     let encoding_key =
         EncodingKey::from_rsa_pem(&ca.pkey.private_key_to_pem_pkcs8().unwrap()).unwrap();
 
-    const EXPIRES_IN_SECS: u64 = 40;
-    const REFRESH_BEFORE_SECS: u64 = 20;
+    const EXPIRES_IN_SECS: u64 = 20;
```
Comment (jkosh44) [inline src/environmentd/tests/auth.rs:813]: In theory we can keep increasing this number to reduce flakiness without affecting test time.

**Comment by jkosh44** [comment]
Do people agree with my assessment of the cause of the test flake?

File: src/environmentd/tests/auth.rs (hunk 1 above)
Comment (jkosh44) [reply src/environmentd/tests/auth.rs:813]: In fact, increasing this to `100000000` locally does not affect the test time.

**Review by mjibson — COMMENTED** [review]
I concur with your diagnosis. This diff would do what it advertises. My one concern is we now no longer test the time between initial token grant and refresh since refresh now happens immediately. That time frame is where most of our active psql sessions are in their auth lifecycle (waiting to start a refresh). I do not think that decreasing test time is worth losing this.

How about: `const REFRESH_BEFORE_SECS: u64 = EXPIRES_IN_SECS - 5;`? If we are currently waiting 20s for each refresh, we'd now be waiting 5s for each refresh, and a 15s gain per refresh would still make the tests much faster, and allow us to sit in the "not yet refreshing" state for a while.

What do you think about removing all the duplication and only having one copy of these consts?

**Comment by jkosh44** [comment]
> How about: `const REFRESH_BEFORE_SECS: u64 = EXPIRES_IN_SECS - 5;`? If we are currently waiting 20s for each refresh, we'd now be waiting 5s for each refresh, and a 15s gain per refresh would still make the tests much faster, and allow us to sit in the "not yet refreshing" state for a while.

I think that's a good compromise, just pushed an update.

> What do you think about removing all the duplication and only having one copy of these consts?

I think that would be cleaner, just pushed an update.

**Review by mjibson — APPROVED** [review]

**Pull request merged**